src/janet.zig

```diff
@@ -2291,3 +2291,62 @@ test "function call" {
     try func.pcall(&[_]Janet{ wrap(i32, 2), wrap(i32, 2) }, &sum, null);
     try testing.expectEqual(@as(i32, 4), try sum.unwrap(i32));
 }
+
+pub const ParserStatus = enum(c_int) {
+    root = 0,
+    @"error" = 1,
+    pending = 2,
+    dead = 3,
+};
+
+pub const Parser = struct {
+    inner: c.JanetParser,
+
+    pub fn init() @This() {
+        var this: @This() = undefined;
+        c.janet_parser_init(&this.inner);
+        return this;
+    }
+    pub fn deinit(this: *@This()) void {
```

Comment:
Can I ask to change this and all other cases to `self: *Parser`? All cases of `this` should probably be converted to `self`. We already use this style in this codebase and iirc Loris Cro did some evangelism on the topic that "`@This()` should be used in metaprogramming when you create types and you don't have the type's name".